ragelib/image_fetcher.py

```diff
@@ -16,15 +17,27 @@ def __init__(self, data, geckodriver_path, logger):
         logger.info(f"Using {geckodriver_path} for geckodriver")
         self.driver = Firefox(executable_path=geckodriver_path, firefox_options=options)
 
-
     def get_graph_screenshot(self, url, driver):
         wait = WebDriverWait(driver, timeout=60)
         self.logger.debug(f"Begun fetching url {url}")
         driver.get(url)
-
-        # On draw starting the progress_img element appears (spinner). When done it disappears.
-        wait.until(expected.visibility_of_element_located((By.ID, 'progress_img')))
-        self.logger.debug("Element #progress_img now visible, graph loading")
+        
+        try:
+            WebDriverWait(driver, timeout=10).until(expected.alert_is_present())

```

Comment:
Does this add a 10s delay to all image fetches, even successful ones?

Reply:
looks like it, yeah

Reply:
Maybe wait until either the popup is present or the graph is visible, can you wait for a or b? And then check which one is present afterwards.